driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -343,14 +350,13 @@ private static List<ServerTuple> atMostNRandom(final ArrayList<ServerDescription
         return result;
     }
 
-    private ServerSelector getCompositeServerSelector(final ServerSelector serverSelector) {
+    private ServerSelector getCompleteServerSelector(final ServerSelector serverSelector, final ServerDeprioritization serverDeprioritization) {
         ServerSelector latencyMinimizingServerSelector =
                 new LatencyMinimizingServerSelector(settings.getLocalThreshold(MILLISECONDS), MILLISECONDS);
-        if (settings.getServerSelector() == null) {
-            return new CompositeServerSelector(asList(serverSelector, latencyMinimizingServerSelector));
-        } else {
-            return new CompositeServerSelector(asList(serverSelector, settings.getServerSelector(), latencyMinimizingServerSelector));
-        }
+        CompositeServerSelector compositeSelector = settings.getServerSelector() == null
+                ? new CompositeServerSelector(asList(serverSelector, latencyMinimizingServerSelector))
+                : new CompositeServerSelector(asList(serverSelector, settings.getServerSelector(), latencyMinimizingServerSelector));
+        return serverDeprioritization.apply(compositeSelector);
```

Comment:
May as well remove the rest of the duplication:

```
List<ServerSelector> selectors = Stream.of(
        serverSelector,
        settings.getServerSelector(), // can be null
        new LatencyMinimizingServerSelector(settings.getLocalThreshold(MILLISECONDS), MILLISECONDS)
).filter(v -> v != null).collect(Collectors.toList());
return serverDeprioritization.apply(new CompositeServerSelector(selectors));
```

Reply:
[Done](https://github.com/mongodb/mongo-java-driver/pull/1367/commits/6574e241434f8c8946984ade559695e2aa9f29b3)